src/data_type/image.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+/**
+ * Abstract Image attribute class.
+ */
+module.exports = class Image {
+  constructor(content) {
+    if (new.target === Image) {
+      throw TypeError('Image is an abstract class, so cannot be instantiated');
+    }
+    this.content = content;
+  }
+
+  /**
+   * Get the raw image content.
+   *
+   * @returns {ByteBuffer}
+   */
+  getContent() {
+    return this.content;
+  }
+
+  /**
+   * Get the base64 image content.
+   *
+   * @returns {string}
+   */
+  getBase64Content() {
+    return `data:${this.getMimeType()};base64,${this.content.toBase64()}`;
+  }
+
+  /**
+   * Abstract method to return the image mime type.
+   *
+   * @returns {string}
+   */
+  getMimeType() {
+    throw new Error(`${this.constructor.name} must implement getMimeType()`);
```

Comment:
This is fine.

But now that I look at it, I'm not sure why the Java SDK does it in the way that it does.

Rather than forcing every subclass to implement the abstract method, we could have a _protected_ constructor on the abstract base class that requires the mimeType is passed in.  Then have a _final getMimeType_ method that returns the value.

Not a big deal.  Might be slightly neater though - if it works in node.js that is.

Reply:
@bucky-boy I've reorganised the code in 1b4694e54a2e574e83769029a97ac22d6b24b071 - I think it's close to achieving your suggestion - does this look better?

cc @echarrod